torch/_numpy/_dtypes.py

```diff
@@ -399,24 +408,22 @@ def issubclass_(arg, klass):
 
 def issubdtype(arg1, arg2):
     # cf https://github.com/numpy/numpy/blob/v1.24.0/numpy/core/numerictypes.py#L356-L420
+    def str_to_abstract(t):
+        if isinstance(t, str) and t in _abstract_dtypes:
+            return globals()[t]
+        return t
+
+    arg1 = str_to_abstract(arg1)
+    arg2 = str_to_abstract(arg2)
```

Comment:
Is this required? It doesn't appear in `np.issubdtype` and calling `np.issubdtype("integer", np.generic)` raises an error:
```python
    415 if not issubclass_(arg1, generic):
--> 416     arg1 = dtype(arg1).type
    417 if not issubclass_(arg2, generic):
    418     arg2 = dtype(arg2).type

TypeError: data type 'integer' not understood
```

Reply:
The issue is that, to avoid having `np.dtype`s in Dyanmo's FX graph, we "serialize" `np.dtype`s using their strings. This is alright for the most part, as constructions like `np.random.randn(3).astype("float")` work, but here we need to work around this very issue in our implementation of this function.

Reply:
`issubdtype` is constant folded though so should never appear directly in an fx graph. Doesn't that mean we can just call it with the original object? Or is the error coming from an `issubdtype` call that's inside another `torch._numpy` call?

Reply:
We could do that, but we would need to reimplement (or better, clean-up) the implementation of NumPy dtypes in dynamo. NumPy dtypes are implemented at the moment via the `as_proxy` call on a NumPy variable:
https://github.com/pytorch/pytorch/blob/3407541b0c58315b6252811eeecc600acf0798b7/torch/_dynamo/variables/misc.py#L834-L848

We could implement them as a `ConstantVariable`, but since we do not want to ever serialize them, I think the current approach is just alright.

Reply:
Okay, in that case I think just update this change with a comment on why we differ from numpy and also it would be good to update the code you linked to say that we're using strings to avoid having `np.dtype` objects in the fx graph.

Reply:
done